src/apis/platformvm/addvotetx.ts

```diff
@@ -0,0 +1,326 @@
+    return {
+      vote: this.vote.serialize(encoding)
+    }
+  }
+
+  deserialize(fields: object, encoding: SerializedEncoding = "hex"): this {
+    this.vote = this.vote.deserialize(fields, encoding)
+    return this
+  }
+
+  getVote(): SimpleVote {
+    return this.vote
+  }
+
+  addVote(optionIndex: number) {
+    const optionIndexBuff = Buffer.alloc(4)
+    optionIndexBuff.writeUInt32BE(optionIndex, 0)
+    this.vote = new SimpleVote(optionIndexBuff)
+  }
+
+  fromBuffer(bytes: Buffer, offset: number = 0): number {
+    // Read the number of length of the following bytes
+    bintools.copyFrom(bytes, offset, offset + 4).readUInt32BE(0)
+    offset += 4
+    // Read the codec version number
+    bintools.copyFrom(bytes, offset, offset + 2).readUInt16BE(0)
+    offset += 2
+    const voteTypeId = bintools
+      .copyFrom(bytes, offset, offset + 4)
+      .readUInt32BE(0)
+    offset += 4
+    switch (voteTypeId) {
+      case PlatformVMConstants.SIMPLEVOTE_TYPE_ID:
+        this.vote = new SimpleVote(bintools.copyFrom(bytes, offset, offset + 4))
+        offset += 4
+        break
+      default:
+        throw `Unsupported vote type: ${voteTypeId}`
+    }
+    return offset
+  }
+
+  toBuffer(): Buffer {
+    const codecVersion = Buffer.alloc(2)
+    codecVersion.writeUInt8(DEFAULT_CAMINOGO_CODEC_VERSION, 0)
+    const typeId = Buffer.alloc(4)
+    typeId.writeUInt32BE(this.vote.getTypeId(), 0)
+    const buff = this.vote.getOptionIndex()
+    const totalByteLength = 2 + typeId.length + buff.length
+    const numLength = Buffer.alloc(4)
+    numLength.writeUInt32BE(totalByteLength, 0)
+    return Buffer.concat(
+      [numLength, codecVersion, typeId, buff],
+      numLength.length + totalByteLength
+    )
+  }
+}
+/**
+ * Class representing an unsigned AddVoteTx transaction.
+ */
+export class AddVoteTx extends BaseTx {
+  protected _typeName = "AddVoteTx"
+  protected _typeID = PlatformVMConstants.ADDVOTETX
+
+  serialize(encoding: SerializedEncoding = "hex"): object {
+    let fields: object = super.serialize(encoding)
+    return {
+      ...fields,
+      voterAddress: serialization.encoder(
+        this.voterAddress,
+        encoding,
+        "Buffer",
+        "cb58"
+      ),
+      voterAuth: this.voterAuth.serialize(encoding),
+      votePayload: this.votePayload.serialize(encoding)
+    }
+  }
+  deserialize(fields: object, encoding: SerializedEncoding = "hex") {
+    super.deserialize(fields, encoding)
+    this.voterAddress = serialization.decoder(
+      fields["voterAddress"],
+      encoding,
+      "cb58",
+      "Buffer",
+      20
+    )
+    this.voterAuth.deserialize(fields["voterAuth"], encoding)
+    this.votePayload = this.votePayload.deserialize(fields, encoding)
+  }
+
+  protected upgradeVersionID = new UpgradeVersionID()
+  protected proposalID = Buffer.alloc(32)
+  protected votePayload: VoteWrapper
+  protected voterAddress = Buffer.alloc(20)
+  protected voterAuth: SubnetAuth
```

Comment:
can you move these properties further up right after typeID  (for easier readability)